examples/samples/SendersReceiversWithQueues.ts

```diff
@@ -0,0 +1,105 @@
+import {
+  ReceiveMode,
+  Namespace,
+  generateUuid,
+  QueueClient,
+  SendableMessageInfo,
+  OnMessage,
+  ServiceBusMessage,
+  OnError,
+  MessagingError,
+  delay,
+  ReceiveHandler
+} from "../../lib";
+import * as dotenv from "dotenv";
+dotenv.config();
+
+const str = process.env.SERVICEBUS_CONNECTION_STRING || "";
+const path = process.env.QUEUE_NAME || "";
+console.log("str: ", str);
+console.log("path: ", path);
+
+let ns: Namespace;
+
+async function sendMessages(queueClient: QueueClient): Promise<void> {
+  const data: { lastName: string; firstName: string }[] = [
+    { lastName: "Einstein", firstName: "Albert" },
+    { lastName: "Heisenberg", firstName: "Werner" },
+    { lastName: "Curie", firstName: "Marie" },
+    { lastName: "Hawking", firstName: "Steven" },
+    { lastName: "Newton", firstName: "Isaac" },
+    { lastName: "Bohr", firstName: "Niels" },
+    { lastName: "Faraday", firstName: "Michael" },
+    { lastName: "Galilei", firstName: "Galileo" },
+    { lastName: "Kepler", firstName: "Johannes" },
+    { lastName: "Kopernikus", firstName: "Nikolaus" }
+  ];
+
+  for (let index = 0; index < data.length; index++) {
+    const element = data[index];
+    const message: SendableMessageInfo = {
+      body: `${element.firstName} ${element.lastName}`,
+      label: "Scientist",
+      timeToLive: 2 * 60 * 1000, // 2 minutes
+      messageId: generateUuid()
+    };
+
+    console.log(`Sending ${message.body}`);
+    await queueClient.send(message);
+  }
+}
+
+async function main(): Promise<void> {
+  ns = Namespace.createFromConnectionString(str);
+  const client = ns.createQueueClient(path, { receiveMode: ReceiveMode.peekLock });
+
+  // populate the queue with messages
+  await sendMessages(client);
+
+  let rcvHandler: ReceiveHandler;
+
+  let received = 0;
+
+  // retrieve all the messages that were sent to the queue (10 messages)
+  const onMessage: OnMessage = async (brokeredMessage: ServiceBusMessage) => {
+    if (received < 10) {
+      console.log(
+        "### Actual message:",
+        brokeredMessage.body ? brokeredMessage.body.toString() : undefined
+      );
+
+      console.log(`Received: ${received + 1}, message ID: ${brokeredMessage.messageId}`);
+      // we received a message
+      received++;
+
```

Comment:
To keep things simple, I would re-order the code inside the `if` block to
- Increment the counter
- Then do the console.log. This way we wont be doing the plus 1 in 2 places. Also, skip the messageId in the logs and include the counter in the same statement as that for the message body